lalrpop/src/grammar/parse_tree.rs

```diff
@@ -149,10 +149,12 @@ pub struct NonterminalData {
     pub alternatives: Vec<Alternative>
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Annotation {
     pub id_span: Span,
     pub id: InternedString,
+    // TODO nixpulvis: This is not exactly the correct type.
+    pub parameters: Vec<(InternedString, InternedString)>,
```

Comment:
Let's add a new struct for this:

```
struct AnnotationKeyValue {
    span: Span,
    key: InternedString,
    value: InternedString,
}
```

Also, we're going to have to check now that annotations like `#[inline]` fit the right form. I wonder if it'd be worth just hard-coding that stuff into the parser, rather than checking it programmatically in `validate`? (We accept so few annotations now.)

Otherwise, we should somehow make the checking more declarative. Heh, I suppose we could make a second grammar just for annotations and then write a custom tokenizer for them that emits tokens (or just parse their Debug output). That'd be...kind of nifty actually.